lib/Frontend/Frontend.cpp

```diff
@@ -517,7 +517,9 @@ void CompilerInstance::performSemaUpTo(SourceFile::ASTStage_t LimitStage) {
   }
 
   FrontendStatsTracer tracer(Context->Stats, "perform-sema");
-  Context->LoadedModules[MainModule->getName()] = getMainModule();
+
+  ModuleDecl *mainModule = getMainModule();
+  Context->LoadedModules[MainModule->getName()] = mainModule;
```

Comment:
Can you read the name from `mainModule` and not the instance variable here as well?

Reply:
If `getMainModule` is inlined it probably would, but it still seems like a funny thing to rely on.

Reply:
It’s not a perf thing, it’s a code smell thing.  It’s lazily initialized so all direct access to the instance variable can break the laziness contract - order of evaluation issues aside.